addons/mail/tests/test_ir_mail_server.py

```diff
@@ -441,3 +444,253 @@ def test_personal_mail_server(self):
+                for i in range(77)
+            ])
+            mails_other = self.env["mail.mail"].create([
+                {'state': 'outgoing', 'email_to': 'target@test.com', 'email_from': user_1.email}
+                for i in range(100)
+            ])
+
+        mails = mails_other + mails_user_1 + mails_user_2
+
+        self.assertEqual(mail_server_1.owner_limit_count, 0)
+        self.assertFalse(mail_server_1.owner_limit_time)
+
+        with self.mock_smtplib_connection(), self._freeze_time("2025-01-01 20:05:23"):
+            mails.send()
+
+        self.assertEqual(mail_server_1.owner_limit_count, 20)
+        self.assertEqual(mail_server_1.owner_limit_time, datetime(2025, 1, 1, 20, 5, 23))
+
+        self.assertEqual(self.connect_mocked.call_count, 3, "Called once for each mail server")
+
+        # Should have sent 20 emails, and delayed the other
+        self.assertEqual(self.env["mail.mail"].search_count(
+            [('id', 'in', mails_user_1.ids), ('state', '=', 'sent')]), 20)
+        delayed = self.env["mail.mail"].search(
+            [('id', 'in', mails_user_1.ids), ('state', '!=', 'sent')])
+        self.assertEqual(len(delayed), 82 - 20)
+        scheduled_dates = delayed.mapped('scheduled_date')
+        self.assertTrue(all(scheduled_dates))
+        scheduled_dates = sorted(scheduled_dates)
+
+        # Check the emails of the first user
+        # First 20 delayed will be sent in 1 minutes
+        # Next 20 delayed will be sent in 2 minutes
+        # ...
+        self.assertEqual(set(scheduled_dates[:20]), {datetime(2025, 1, 1, 20, 6, 23)})
+        self.assertEqual(set(scheduled_dates[20:40]), {datetime(2025, 1, 1, 20, 7, 23)})
+        self.assertEqual(set(scheduled_dates[40:60]), {datetime(2025, 1, 1, 20, 8, 23)})
+        self.assertEqual(set(scheduled_dates[60:80]), {datetime(2025, 1, 1, 20, 9, 23)})
+
+        # Check the emails of the second user
+        # Should have sent 20 emails, and delayed the other
+        self.assertEqual(self.env["mail.mail"].search_count(
+            [('id', 'in', mails_user_2.ids), ('state', '=', 'sent')]), 20)
+        delayed = self.env["mail.mail"].search(
+            [('id', 'in', mails_user_2.ids), ('state', '!=', 'sent')])
+        self.assertEqual(len(delayed), 77 - 20)
+        scheduled_dates = delayed.mapped('scheduled_date')
+        self.assertTrue(all(scheduled_dates))
+        scheduled_dates = sorted(scheduled_dates)
+
+        self.assertEqual(set(scheduled_dates[:20]), {datetime(2025, 1, 1, 20, 6, 23)})
+        self.assertEqual(set(scheduled_dates[20:40]), {datetime(2025, 1, 1, 20, 7, 23)})
+        self.assertEqual(set(scheduled_dates[40:]), {datetime(2025, 1, 1, 20, 8, 23)})
+
+        # Check that the email not related to personal mail server are all sent
+        self.assertEqual(set(mails_other.mapped('state')), {'sent'})
+
+        # The user 1 continues to send emails
```

Comment:
I think we should use perfect present tense (important :D )
```suggestion
        # User 1 continues sending emails
```